dask_sql/physical/rex/core/call.py

```diff
@@ -699,11 +706,23 @@ def __init__(self):
         super().__init__(self.datetime_sub)
 
     def datetime_sub(self, unit, df1, df2):
+        if is_timestamp_nano(df1):
+            df1 = df1 * 10**9
+        if is_timestamp_nano(df2):
+            df2 = df2 * 10**9
+        if "datetime64[s]" == str(getattr(df1, "dtype", "")):
+            df1 = df1.astype("datetime64[ns]")
+        if "datetime64[s]" == str(getattr(df2, "dtype", "")):
+            df2 = df2.astype("datetime64[ns]")
```

Comment:
Getting a little tied up in the logic here, but think we might also want to scale the `result` based on the values of `is_timestamp_nano(df1|df2)`?

Reply:
I'm not really sure what you mean here? Whenever we do a subtraction operation (like with `timestampdiff`), this extra logic is to ensure uniformity so that we're working in nanoseconds.